src/h3lib/include/polygonAlgos.h

```diff
@@ -92,6 +92,13 @@ bool GENERIC_LOOP_ALGO(pointInside)(const TYPE *loop, const BBox *bbox,
             b = tmp;
         }
 
+        // If the latitude matches exactly, we'll hit an edge case where
+        // the ray passes through the vertex twice on successive segment
+        // checks. To avoid this, adjust the latiude northward if needed
+        if (lat == a.lat || lat == b.lat) {
+            lat += DBL_EPSILON;
+        }
```

Comment:
What happens if the latitude is maximally north already? I assume there won't be any issue here?

Reply:
You may be correct - if we're checking the latlng point `[90, 0]` (i.e. the north pole), this algorithm will I think fail to contain it in any polygon. But because we're using Cartesian polygons to begin with, there's no pole-enclosing polygon anyway. This is all moot for the `polyfill` usage, however, because the north pole cell at res 15 has a center latitude of `89.99999581738042`, so it's not possible for this issue to arise. If we offered point-in-poly as an external function, this might be an issue, but probably one we could address in documentation.

Reply:
So the `lat` and `lng` here is provided by the caller, so this is theoretically a problem, but we just use the centerpoint of the polygon in reality: https://github.com/uber/h3/blob/ba89a19aa6d765ceb6f52d0e956b576293a67911/src/h3lib/lib/algos.c#L1004

There might be a problem with `pointInsideLinkedGeoLoop`, though, since it just uses the first point which could be the northernmost one: https://github.com/uber/h3/blob/adde6daaf96762175c73df24ad8ec6afcc0e4db5/src/h3lib/lib/linkedGeo.c#L191

Reply:
The `pointInsideLinkedGeoLoop` function is looking at whether one loop is inside another. So for this issue to occur, you'd need a fairly pathological shape, where not only was one vertex of the outer loop on the north pole, but one vertex of the inner loop was also on the north pole, eg.

<img width="259" alt="image" src="https://user-images.githubusercontent.com/163940/164570986-ddc2cc1c-b02a-48ca-bf70-8a9151dce5e0.png">

But again, the loops involved here are derived from H3 cell boundaries, and there is **no cell** with a center or a vertex on the north pole.

Reply:
Added a long comment noting the issue.